src/main/java/org/jenkinsci/plugins/ghprb/extensions/status/GhprbSimpleQueueStatusUpdater.java

```diff
@@ -0,0 +1,166 @@
+        List<Queue.Item> items = queue.getApproximateItemsQuickly();
+        
+        HashMap<Label, Integer> queueLengthsPerLabel = new HashMap<Label, Integer>();
+        ArrayList<UpdateEntry> updateEntries = new ArrayList<UpdateEntry>();
+
+        // Walk the queue items, finding those that are launched by the ghprb.
+        // Note that this is a snapshot.  We want to do our best to avoid
+        // incorrect updates, etc.
+        for (Queue.Item item : items) {
+            // 
+            // Grab the position in the queue for the particular label.
+            // The positions we are recording here are reversed, so we'll
+            // switch the position vs. the maximum queueLength for each label
+            // when we update the status.
+            Label assignedLabel = item.getAssignedLabel();
+            
+            if (assignedLabel == null) {
+                continue;
+            }
+            
+            int queuePosition = 0;
+            if (queueLengthsPerLabel.containsKey(assignedLabel)) {
+                queuePosition = queueLengthsPerLabel.get(assignedLabel);
+            }
+            // Find a GhprbUpdateQueueStatus item
+            GhprbUpdateQueueStatus updateStatusAction = item.getAction(GhprbUpdateQueueStatus.class);
+            // Grab the position in the queue for the particular label.
+            // The positions we are recording here are reversed, so we'll
+            // switch the position vs. the maximum queueLength for each label
+            // when we update the status.
+            queueLengthsPerLabel.put(assignedLabel, queuePosition + 1);
+            
+            // If it's not GHPRB launched, go on to the next queue item.
+            if (updateStatusAction == null) {
+                continue;
+            }
+
+            // We don't keep track of the GHRepository connection on the queue item for simplicities
+            // sake, so if there isn't a GHRepository, continue. This could happen on restart for instance.
+
+            if (updateStatusAction.getRepository() == null) {
+                continue;
+            }
+            
+            updateEntries.add(new UpdateEntry(item, updateStatusAction, queuePosition + 1));
+        }
+        
+        // Run the actual update
+        for (UpdateEntry updateItem : updateEntries) {
+            GhprbUpdateQueueStatus updateStatusAction = updateItem.getStatusAction();
+            Queue.Item queueItem = updateItem.getItem();
+            int queuePosition = updateItem.getPosition();
+            
+            try {
+                synchronized(updateStatusAction) {
+                    if (updateStatusAction.getStarted()) {
+                        continue;
+                    }
+
+                    int queueLength = queueLengthsPerLabel.get(queueItem.getAssignedLabel());
+                    queuePosition = queueLength - queuePosition + 1;
+                    StringBuilder sb = new StringBuilder();
```

Comment:
can you move that string building into its own method? 

And if the string is rather static you can just construct a string, instead of using a stringbuilder, there is no real advantage of using a string builder here.